src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/latest/test_resource.py

```diff
@@ -791,14 +803,25 @@ def test_resource_policyset_default(self, resource_group):
     @ResourceGroupPreparer(name_prefix='cli_test_policyset_management_group')
     @AllowLargeResponse()
     def test_resource_policyset_management_group(self, resource_group):
-        self.resource_policyset_operations(resource_group, 'AzGovTest8')
+        management_group_name = self.create_random_name('cli-test-mgmt-group', 30)
+        self.cmd('account management-group create -n ' + management_group_name)
+        try:
+            self.resource_policyset_operations(resource_group, management_group_name)
```

Comment:
Where in `resource_policyset_operations` is the test usually failing? 
Using a try-catch really just bypasses the problems within the test, making the test a lot less useful.
This may also be a candidate for a `record_only`

Reply:
it is a `finally` block so that i can clean up the created management group, not exceptions get swallowed. the right way might be to create a management group preparer which is a bit heavy.

Reply:
Ah, misread, makes sense.